src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java

```diff
@@ -21,73 +21,65 @@
 package org.rumbledb.cli;
 
 import org.antlr.v4.runtime.BailErrorStrategy;
-import org.antlr.v4.runtime.CharStream;
 import org.antlr.v4.runtime.CharStreams;
 import org.antlr.v4.runtime.CommonTokenStream;
 import org.antlr.v4.runtime.misc.ParseCancellationException;
 import org.apache.hadoop.fs.FSDataInputStream;
-import org.apache.hadoop.fs.FSDataOutputStream;
-import org.apache.hadoop.fs.Path;
 import org.apache.spark.api.java.JavaRDD;
 import org.rumbledb.api.Item;
 import org.rumbledb.compiler.TranslationVisitor;
 import org.rumbledb.compiler.VisitorHelpers;
 import org.rumbledb.config.RumbleRuntimeConfiguration;
 import org.rumbledb.context.DynamicContext;
+import org.rumbledb.exceptions.CannotRetrieveResourceException;
 import org.rumbledb.exceptions.ExceptionMetadata;
 import org.rumbledb.exceptions.OurBadException;
 import org.rumbledb.exceptions.ParsingException;
 import org.rumbledb.expressions.module.MainModule;
 import org.rumbledb.parser.JsoniqLexer;
 import org.rumbledb.parser.JsoniqParser;
 import org.rumbledb.runtime.RuntimeIterator;
+import org.rumbledb.runtime.functions.input.UrlValidator;
 
 import sparksoniq.spark.SparkSessionManager;
-import sparksoniq.utils.FileUtils;
-
-import java.io.BufferedOutputStream;
-import java.io.File;
 import java.io.IOException;
-import java.net.URI;
-import java.nio.charset.Charset;
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 
 public class JsoniqQueryExecutor {
-    public static final String TEMP_QUERY_FILE_NAME = "Temp_Query";
     private RumbleRuntimeConfiguration configuration;
-    private boolean useLocalOutputLog;
 
-    public JsoniqQueryExecutor(boolean useLocalOutputLog, RumbleRuntimeConfiguration configuration) {
+    public JsoniqQueryExecutor(RumbleRuntimeConfiguration configuration) {
         this.configuration = configuration;
-        this.useLocalOutputLog = useLocalOutputLog;
         SparkSessionManager.COLLECT_ITEM_LIMIT = configuration.getResultSizeCap();
     }
 
-    public void runLocal(String queryFile, String outputPath) throws IOException {
-        File outputFile = null;
+    private void checkOutputFile(String outputPath) throws IOException {
         if (outputPath != null) {
-            outputFile = new File(outputPath);
-            if (outputFile.exists()) {
+            if (UrlValidator.exists(outputPath, new ExceptionMetadata(0, 0))) {
                 if (!this.configuration.getOverwrite()) {
                     System.err.println(
                         "Output path " + outputPath + " already exists. Please use --overwrite yes to overwrite."
                     );
                     System.exit(1);
```

Comment:
The Hadoop FileContext class will overwrite, no need to delete any more.

Reply:
Cleanup on this file looks amazing